web3.js/src/connection.ts

```diff
@@ -2865,40 +2902,81 @@ export class Connection {
       }
     });
 
-    let timeoutMs = this._confirmTransactionInitialTimeout || 60 * 1000;
-    switch (subscriptionCommitment) {
-      case 'processed':
-      case 'recent':
-      case 'single':
-      case 'confirmed':
-      case 'singleGossip': {
-        timeoutMs = this._confirmTransactionInitialTimeout || 30 * 1000;
-        break;
+    const checkBlockHeight = async () => {
+      const blockHeight = await this.getBlockHeight(commitment);
+      return blockHeight;
+    };
+
+    const expireTx = new Promise<{
+      __type: TransactionStatus.EXPIRED | TransactionStatus.TIMED_OUT;
+      reason: 'blockheight expired' | 'timeout elapsed';
+    }>((resolve, reject) => {
```

Comment:
Since you went with two different types, you can do this and eliminate the need for a `reason` field!

```suggestion
    const expireTx = new Promise<
      {__type: TransactionStatus.BLOCKHEIGHT_EXCEEDED} | {__type: TransactionStatus.TIMED_OUT}
    >((resolve, reject) => {
```

Reply:
Done!